api/test/opentelemetry/baggage/propagation/text_map_propagator_test.rb

```diff
@@ -184,5 +184,9 @@ def assert_value(context, key, value, metadata = nil)
   entry = OpenTelemetry::Baggage.raw_entries(context: context)[key]
   _(entry).wont_be_nil
   _(entry.value).must_equal(value)
-  _(entry.metadata).must_equal(metadata)
+  if metadata.nil?
+    _(entry.metadata).must_be_nil
```

Comment:
Am I correct that this change is entirely unrelated to the changes to the text map propagator?  It seems this is addressing the mini test complaints about assert_equal on nil values.

Reply:
Yes, exactly. That's why it's in a separate commit.